src/client/src/orangehrmRecruitmentPlugin/components/VacancyCard.vue

```diff
@@ -0,0 +1,191 @@
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+ -->
+
+<template>
+  <div class="orangehrm-paper-container">
+    <div class="orangehrm-vacancy-card-header">
+      <oxd-text type="card-title">
+        {{ vacancyTitle }}
+      </oxd-text>
+      <oxd-button
+        :label="$t('general.apply')"
+        class="oxd-button--success"
+      ></oxd-button>
+    </div>
+    <oxd-divider v-show="vacancyDescription"></oxd-divider>
+    <div :class="{'orangehrm-vacancy-card-body': isViewDetails}">
+      <oxd-text type="toast-message"
+        ><pre
+          v-if="vacancyDescription"
+          class="orangehrm-vacancy-card-pre-tag"
+          >{{ vacancyDescription }}</pre
+        ></oxd-text
+      >
+    </div>
+    <div
+      v-if="vacancyDescription && vacancyDescription.length > descriptionLength"
+      class="orangehrm-vacancy-card-footer"
+    >
+      <a @click="viewDetails">
+        <oxd-text
+          class="orangehrm-vacancy-card-anchor-tag"
+          type="toast-message"
+        >
+          {{
+            isViewDetails
+              ? $t('recruitment.show_more')
+              : $t('recruitment.show_less')
+          }}
+        </oxd-text>
+      </a>
+    </div>
+  </div>
+</template>
+
+<script>
+import OxdDivider from '@ohrm/oxd/core/components/Divider/Divider';
+import useResponsive from '@ohrm/oxd/composables/useResponsive';
+import {toRefs} from 'vue';
+
+export default {
+  name: 'VacancyCard',
+  components: {
+    'oxd-divider': OxdDivider,
+  },
+  props: {
+    vacancyId: {
+      type: Number,
+      required: true,
+    },
+    vacancyTitle: {
+      type: String,
+      required: true,
+    },
+    vacancyDescription: {
+      type: String,
+      required: true,
+    },
+  },
+  setup() {
+    const responsiveState = useResponsive();
+    return {
+      ...toRefs(responsiveState),
+    };
+  },
+  data() {
+    return {
+      viewMore: true,
+      descriptionLength: 0,
+    };
+  },
+  computed: {
+    isViewDetails() {
+      return this.viewMore;
+    },
+    isMobile() {
+      return this.windowWidth < 600;
+    },
+  },
+  watch: {
+    isMobile: function() {
+      this.validDescriptionLength();
+    },
+  },
+  beforeMount() {
+    this.onResize();
+    this.validDescriptionLength();
+    window.addEventListener('resize', this.onResize);
+  },
+  beforeUnmount() {
+    window.removeEventListener('resize', this.onResize);
+  },
+  methods: {
+    viewDetails() {
+      return (this.viewMore = !this.viewMore);
+    },
+    validDescriptionLength() {
+      return this.isMobile
+        ? (this.descriptionLength = 150)
+        : this.windowWidth < 1920
+        ? (this.descriptionLength = 300)
+        : (this.descriptionLength = 400);
```

Comment:
This is not good. based on language width of a character may change.

@Super-Chama can't we use `import useResponsive from '@ohrm/oxd/composables/useResponsive';`
can't we handle this with text wrapping, but I'm not sure since he used pre tag

Reply:
we can. i don't understand how he have done this. will have a call

Reply:
resolved for now